From bbc53fb32111e0b35f98a84e200477721aabf53b Mon Sep 17 00:00:00 2001
From: Putin Lee <putin.li@rock-chips.com>
Date: Thu, 29 Mar 2018 21:37:23 +0800
Subject: [PATCH] video/rockchip: rga2: Fix up dst vir address not 4k align
 will cause crash

If dst buf not 4k align vir address, mmu page may calculate error.
This situation will cause rga hardware not respond, cannot return irq_handle.

Change-Id: I29a104984415b6736713073684c2551252c975b9
Signed-off-by: Putin Lee <putin.li@rock-chips.com>
---
 drivers/video/rockchip/rga2/rga2_mmu_info.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/rockchip/rga2/rga2_mmu_info.c b/drivers/video/rockchip/rga2/rga2_mmu_info.c
index 72ff0099e356..97cb671f5d0a 100644
--- a/drivers/video/rockchip/rga2/rga2_mmu_info.c
+++ b/drivers/video/rockchip/rga2/rga2_mmu_info.c
@@ -468,6 +468,7 @@ static int rga2_mmu_info_BitBlt_mode(struct rga2_reg *reg, struct rga2_req *req)
 						 req->dst.vir_w,
 						 req->dst.vir_h,
 						 &DstStart);
+		DstPageCount = (DstPageCount + 3) & (~3);
 		if (DstPageCount == 0)
 			return -EINVAL;
 	}
-- 
2.35.3

